lib/site_setting_extension.rb

```diff
@@ -176,7 +176,13 @@ def client_settings_json_uncached
   end
 
   # Retrieve all settings
-  def all_settings(include_hidden: false)
+  def all_settings(
+    include_hidden: false,
+    include_locale_setting: true,
+    only_overridden: false,
+    filter_categories: nil,
+    filter_plugin: nil
+  )
```

Comment:
This is getting a bit beefy now. Let's start thinking about how to handle the complexity. (In another PR. 🙂)

![beefcake](https://github.com/discourse/discourse/assets/5259935/f3d1478d-4549-4942-b604-1e10aa67d54b)